CMakeLists.txt

```diff
@@ -16,14 +16,15 @@ list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/CMakeModules/Modules)
 ###############################################################################
 
 # Valid applications and choices
-list(APPEND VALID_APPS ATM ATMW S2S S2SW DATM DATM_NEMS)
+list(APPEND VALID_APPS ATM ATMAERO ATMW S2S S2SW DATM DATM_NEMS)
```

Comment:
I tried the same in the beginning, adding a new application `ATMAERO` or whatever you want to call it. The problem was the following. If you specify a list of suites for CCPP to compile, and you choose suites containing chemistry code as in https://github.com/noaa-gsl/ufs-weather-model/tree/gsl/develop-chem, then you need to compile `APP=ATMAERO`. If you use the functionality to not specify a suite, hence CCPP compiles all suites it can find, then you also need to use `APP=ATMAERO`, because choosing `APP=ATM` won't compile the chemistry code.

It's ok to do that one time, but if similar additions are made in the future for other components, we'll need some dedicated logic (or a catch-all APP) that compiles everything it knows about in order to provide whatever may be needed by any of the CCPP suites.

Reply:
@climbfuji I might be missing something here. Chemistry-aware physics schemes enable chemistry computations conditionally (`cplchm=.true.`). If you compile a list of CCPP suites, some of which are chemistry-aware, the model can still build with `APP=ATM` and run as long as `cplchm=.false.`.

Reply:
Not exactly, the dependencies of the CCPP schemes will be required (it will look for the `.mod` Fortran module files and objects in the library) for every scheme in the list of suites, regardless of whether `cplchm` is true or false.

Reply:
@climbfuji - do you have an example of a configuration that will break as described? I'd like to understand more about this issue.

Reply:
You need the CCPP physics/chemistry code in the gsl/develop-chem branch, which depends on the GOCART library. The easiest way to demonstrate is using the gsl/develop-chem branch of the ufs-weather-model and define an APP like you did. Then compile the code with all suites enabled using `APP=ATM` - it will fail.

Note that you can only use GNU for compiling all suites, since Intel gives up halfway when it tries to optimize the auto-generated caps (even in DEBUG mode).

I don't think this is real show stopper, but we should either abandon the idea to compile all suites (works with GNU only anyway), or we need to define a catch-all=compile-all APP in case no suites are specified. The former seems to be easier.

Reply:
@climbfuji - Thanks, I see your point now. The chemistry process library won't be built if `APP=ATM` while a given CCPP suite may depend on such process library.

Ideally the CCPP parser should trigger the build, perhaps checking for additional metadata. I understand this may require a few changes, since currently the process library is enabled by setting `GOCART=ON` in the top-level CMakeLists.txt file.